packages/cli/src/commands/onboard/command.ts

```diff
@@ -50,7 +60,11 @@ export async function command(): Promise<void> {
     await integrations(providerInfo);
   }
 
-  if (!shouldSetupAuth && !shouldSetupScaffolder) {
+  if (shouldDiscoverEntities) {
+    await discover(providerInfo);
+  }
+
+  if (!shouldSetupAuth && !shouldSetupScaffolder && !shouldDiscoverEntities) {
```

Comment:
Starting to feel that maybe this check could have been at the top, with an early return. No rush to change that though.